.drone.yml

```diff
@@ -1,185 +1,244 @@
-      environment: cs-qa
-
-  deploy-to-wcs-qa:
-    image: quay.io/ukhomeofficedigital/kd:v1.16.0
-    environment:
-      - ENVIRONMENT=wcs-qa
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_templates_wcs_qa
-    commands:
-      - source version.txt
-      - echo $VERSION
-      - cd kube-hocs-templates
-      - ./deploy.sh
-    when:
-      event: deployment
-      environment: wcs-qa
-
-  deploy-to-hocs-qax:
-    image: quay.io/ukhomeofficedigital/kd:v1.16.0
-    environment:
-      - ENVIRONMENT=hocs-qax
-      - KUBE_SERVER=https://kube-api-notprod.notprod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_templates_qax
-    commands:
-      - source version.txt
-      - echo $VERSION
-      - cd kube-hocs-templates
-      - ./deploy.sh
-    when:
-      event: deployment
-      environment: hocs-qax
-
-  deploy-to-cs-prod:
-    image: quay.io/ukhomeofficedigital/kd:v1.16.0
-    environment:
-      - ENVIRONMENT=cs-prod
-      - KUBE_SERVER=https://kube-api-prod.prod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_templates_cs_prod
-    commands:
-      - cd kube-hocs-templates
-      - ./deploy.sh
-    when:
-      event: deployment
-      environment: cs-prod
-
-  deploy-to-wcs-prod:
-    image: quay.io/ukhomeofficedigital/kd:v1.16.0
-    environment:
-      - ENVIRONMENT=wcs-prod
-      - KUBE_SERVER=https://kube-api-prod.prod.acp.homeoffice.gov.uk
-    secrets:
-      - hocs_templates_wcs_prod
-    commands:
-      - cd kube-hocs-templates
-      - ./deploy.sh
-    when:
-      event: deployment
-      environment: wcs-prod
+---
+
+kind: pipeline
+type: kubernetes
+name: build
+
+steps:
+- name: build project
+  image: quay.io/ukhomeofficedigital/openjdk11:v11.0.5_10
+  commands:
+  - ./gradlew build
+  when:
+    event:
+      - push
+
+- name: sonar scanner
+  image: quay.io/ukhomeofficedigital/sonar-scanner:v3.0.3
+  when:
+    event:
+      - push
+  depends_on:
+    - build project
+
+- name: build & push
+  image: plugins/docker
+  settings:
+    registry: quay.io
+    repo: quay.io/ukhomeofficedigital/hocs-templates
+    tags:
+      - build_${DRONE_BUILD_NUMBER}
+      - ${DRONE_COMMIT_SHA}
+      - branch-${DRONE_COMMIT_BRANCH/\//_}
+  environment:
+    DOCKER_PASSWORD:
+      from_secret: QUAY_ROBOT_PASSWORD
+    DOCKER_USERNAME: ukhomeofficedigital+hocs_quay_robot
+  when:
+    event:
+      - push
+  depends_on:
+    - build project
+
+- name: build & push latest
+  image: plugins/docker
+  settings:
+    registry: quay.io
+    repo: quay.io/ukhomeofficedigital/hocs-templates
+    tags:
+      - master
+  environment:
+    DOCKER_PASSWORD:
+      from_secret: QUAY_ROBOT_PASSWORD
+    DOCKER_USERNAME: ukhomeofficedigital+hocs_quay_robot
+  when:
+    branch:
+      - master
+    event:
+      - push
+  depends_on:
+    - build project
+
+trigger:
+  event:
+    - push
+
+---
+
+kind: pipeline
+type: kubernetes
+name: deploy
+depends_on:
+  - build
```

Comment:
Are the kind and type required again?

Reply:
`type` is definitely required. `kind` is boilerplate I've seen on every example, so I assume it's also required.

Reply:
Thanks for verifying.